src/tools/db-upgrade/DbUpgrader.cpp

```diff
@@ -0,0 +1,461 @@
+
+                lastTagId = tagId;
+                lastVertexId = vId;
+            } else if (NebulaKeyUtilsV1::isEdge(key)) {
+                auto svId = NebulaKeyUtilsV1::getSrcId(key);
+                auto edgetype = NebulaKeyUtilsV1::getEdgeType(key);
+                auto ranking = NebulaKeyUtilsV1::getRank(key);
+                auto dvId = NebulaKeyUtilsV1::getDstId(key);
+                auto version = NebulaKeyUtilsV1::getVersion(key);
+
+                auto it = edgeSchemas_.find(std::abs(edgetype));
+                if (it == edgeSchemas_.end()) {
+                    // Invalid data
+                    iter->next();
+                    continue;
+                }
+                if (svId == lastSrcVertexId &&
+                    edgetype == lastEdgeType &&
+                    ranking == lastRank &&
+                    dvId == lastDstVertexId) {
+                    // Multi version
+                    iter->next();
+                    continue;
+                }
+
+                // auto strsvId = folly::to<std::string>(svId);
+                // auto strdvId = folly::to<std::string>(dvId);
+                auto strsvId = std::string(reinterpret_cast<const char*>(&svId), sizeof(svId));
+                auto strdvId = std::string(reinterpret_cast<const char*>(&dvId), sizeof(dvId));
+                if (!isValidVidLen(strsvId, strdvId)) {
+                    iter->next();
+                    continue;
+                }
+
+                auto newEdgeSchema = it->second.back();
+
+                // Generate 2.0 key
+                auto newKey = NebulaKeyUtils::edgeKey(spaceVidLen_,
+                                                      partId,
+                                                      strsvId,
+                                                      edgetype,
+                                                      ranking,
+                                                      strdvId,
+                                                      version);
+                auto val = iter->val();
+                auto reader = RowReaderWrapper::getEdgePropReader(schemaMan_.get(),
+                                                                  spaceId_,
+                                                                  std::abs(edgetype),
+                                                                  val);
+                if (!reader) {
+                    LOG(INFO) << "Can't get edge reader of " << edgetype;
+                    iter->next();
+                    continue;
+                }
+                // Generate 2.0 value and index records
+                encodeEdgeValue(partId, reader.get(), newEdgeSchema.get(), newKey, strsvId,
+                                edgetype, ranking, strdvId, data);
+                lastSrcVertexId = svId;
+                lastEdgeType  = edgetype;
+                lastRank = ranking;
+                lastDstVertexId = dvId;
+            }
```

Comment:
Two type is missed, kUUID and kSystem, write them as usual

Reply:
Good，kSystem is need in RocksEngine::allParts， so kSystem in  1.0 and 2.0   are same
Here we will handle kUUID